portfolio/src/main/webapp/scripts/snake/sketch.js

```diff
@@ -0,0 +1,283 @@
+/** Game board */
+let cnv;
+/** Game board container in HTML */
+let cnvDiv;
+/** 4 sections on canvas to represent grids for 4-Snake */
+let sec1;
+let sec2;
+let sec3;
+let sec4;
+/** Scale of the game (# of grids = (width * height / GRID_SCALE^2))*/
+const GRID_SCALE = 20;
+
+/** Map selected game difficulty to framerate to change the "speed" of the game */
+const difficultyMap = {
+    'easy' : 10,
```

Comment:
Consider using enums as the keys here (and in corresponding usage in template) to avoid potential mismatch

Reply:
This is a good point but I'm not 100% sure if I understood exactly what you were getting at cause I don't think enums are native to JS. I pushed a change wherein I call ```Object.freeze(Difficulty)``` in order to simulate the entire thing being an enum and where I'm using it, I call a switch statement on the value received from the difficulty select option in the html. Did you mean I should be passing an enum itself from the HTML template as the value in the select option?

Reply:
Ah sorry stuck in the typescript mindset :) I'm not familiar with Object.freeze() but quickly glimpsing at it that seems like the technically accurate thing to do, though you could get away with using this without that call I believe.  I did mean to use these as well to the select option values so you could do away with all this case switching and easily keep it consistent between the two!

Reply:
Ah I see, yeah Object.freeze feels a bit unnecessary but also it doesn't hurt I guess. Hm yeah I looked more into passing enums from HTML to JS without a framework and I couldn't find much, just this SO post on how to do it in Angular2: https://stackoverflow.com/questions/42464367/angular2-use-enum-value-in-html-value-attribute. I'm thinking it might not be possible in native JS but I could definitely be wrong haha.